### PR TITLE
Move jq-coded builtins to non-C file (fix #424)

### LINKED ISSUE #424: jq-coded builtins should live in a separate, non-C file
It'd be nice if there was no need to escape double-quotes for C, and if the only thing to do to add jq-coded builtins was to append them to a file, then add docs and tests.

### PR DESCRIPTION
**Pull request by dtolnay**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -86,6 +86,10 @@ src/version.h: .remake-version-h
 	$(AM_V_GEN) $(generate_ver); echo "$$ver" > $@
 src/main.c: src/version.h
 
+src/builtin.inc: src/builtin.jq
+	$(AM_V_GEN) sed -e 's/\\/\\\\/g' -e 's/"/\\"/g' -e 's/^/"/' -e 's/$$/\\n"/' $^ > $@
```
Comment (nicowilliams) [inline Makefile.am]: Why `s/$$/../`, why not `s/$/...`?

File: src/builtin.c [hunk 2]
```diff
@@ -1698,10 +1413,8 @@ int builtins_bind(jq_state *jq, block* bb) {
     block_free(*bb);
     return nerrors;
   }
-  for (int i=(int)(sizeof(jq_builtins)/sizeof(jq_builtins[0]))-1; i>=0; i--) {
-    nerrors = builtins_bind_one(jq, bb, jq_builtins[i]);
-    assert(!nerrors);
-  }
+  nerrors = builtins_bind_one(jq, bb, jq_builtins);
```
Comment (nicowilliams) [inline src/builtin.c]: Excellent!

**Comment by nicowilliams** [comment]
I only have one comment (see diffs).

I have a half-baked commit lying around that looks a lot like this, but with the addition of supporting multiple builtin modules (which can wait for some other day; the idea is that users could `include jq "1.4";`, which would allow us to start to make backwards-incompatible changes to the builtins (including `|=` and such syntax that depends on jq-coded helpers) comfortably.

**Comment by nicowilliams** [comment]
Other than my comment about that `sed`, this LGTM.

File: Makefile.am (hunk 1 above)
Comment (dtolnay) [inline Makefile.am]: https://www.gnu.org/software/make/manual/html_node/Variables-in-Recipes.html

> if you want a dollar sign to appear in your recipe, you must double it (‘$$’)

**Comment by dtolnay** [comment]
Rebased as 143e8dc.

File: Makefile.am (hunk 1 above)
Comment (nicowilliams) [inline Makefile.am]: Oh, duh, it's make, and I brainfarted.  Sorry.

**Pull request closed without merging**